test/arcade/core/util/MiniBoxTest.java

```diff
@@ -333,28 +333,64 @@ public void filter_multipleCodes_returnsFiltered() {
         MiniBox box = new MiniBox();
         String code1 = randomString();
         String code2 = randomString();
+        String subcode = randomString();
         String key1 = randomString();
         String key2 = randomString();
         String key3 = randomString();
+        String key4 = randomString();
         String value1 = randomString();
         String value2 = randomString();
         String value3 = randomString();
+        String value4 = randomString();
 
         box.put(key1, value1);
         box.put(code1 + TAG_SEPARATOR + key2, value2);
         box.put(code2 + TAG_SEPARATOR + key3, value3);
+        box.put(code2 + TAG_SEPARATOR + subcode + TAG_SEPARATOR + key4, value4);
 
         ArrayList<String> filteredKeys = new ArrayList<>();
         filteredKeys.add(key3);
+        filteredKeys.add(subcode + TAG_SEPARATOR + key4);
 
         HashMap<String, String> filteredMap = new HashMap<>();
         filteredMap.put(key3, value3);
+        filteredMap.put(subcode + TAG_SEPARATOR + key4, value4);
 
         MiniBox filtered = box.filter(code2);
         assertEquals(filteredKeys, filtered.keys);
         assertEquals(filteredMap, filtered.contents);
     }
 
+    @Test
+    void filter_codeNotFirst_returnsEmpty() {
+        MiniBox box = new MiniBox();
+        String code1 = randomString();
+        String subcode = randomString();
+        String value1 = randomString();
+
+        box.put(code1 + TAG_SEPARATOR + subcode, value1);
+        ArrayList<String> filteredKeys = new ArrayList<>();
+        HashMap<String, String> filteredMap = new HashMap<>();
+
+        MiniBox filtered = box.filter(subcode);
+        assertEquals(filteredKeys, filtered.keys);
```

Comment:
I think:

```java
assertTrue(filtered.keys.isEmpty());
assertTrue(filtered.contents.isEmpty());
```

is more clear about the resulting fields being empty, rather than comparing them to empty instances. Same for the other test.